test/yamlSchemaService.test.ts

```diff
@@ -67,6 +67,50 @@ describe('YAML Schema Service', () => {
       expect(schema.schema.type).eqls('array');
     });
 
+    it('should handle schemas that use draft-04', async () => {
+      const content = `openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: Minimal ping API server
+paths:
+  /ping:
+    get:
+      responses:
+        '200':
+          description: pet response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Pong'
+components:
+  schemas:
+    # base types
+    Pong:
+      type: object
+      required:
+        - ping
+      properties:
+        ping:
+          type: string
+          example: pong`;
+
+      const yamlDock = parse(content);
+      // eslint-disable-next-line @typescript-eslint/no-var-requires
+      const openapiV3Schema = require(path.join(__dirname, './fixtures/sample-openapiv3.0.0-schema.json'));
+
+      requestServiceMock = sandbox.fake.resolves(JSON.stringify(openapiV3Schema));
+      const service = new SchemaService.YAMLSchemaService(requestServiceMock);
+      service.registerCustomSchemaProvider(() => {
+        return new Promise<string | string[]>((resolve) => {
+          resolve('http://fakeschema.faketld');
+        });
+      });
```

Comment:
I came up with this to enforce the retrieval of the schema. Not sure if this is the best, feel free to suggest better ways.